saleor/core/jwt.py

```diff
@@ -103,6 +108,49 @@ def get_user_from_payload(payload: Dict[str, Any]) -> Optional[User]:
 
 def get_user_from_access_token(token: str) -> Optional[User]:
     payload = jwt_decode(token)
-    if payload["type"] != JWT_ACCESS_TYPE:
+    if payload["type"] not in [JWT_ACCESS_TYPE, JWT_THIRDPARTY_ACCESS_TYPE]:
         return None
-    return get_user_from_payload(payload)
+    permission_limits = payload.get(PERMISSION_LIMITS_FIELD, None)
+    user = get_user_from_payload(payload)
+    if permission_limits is not None:
+        user.permission_limits = permission_limits  # type: ignore
```

Comment:
I don't like too much having a magic field, sometimes it can be there, sometimes not.

Could we have a default value in `User` and comment it?

Reply:
Or call the field `effective_permissions` and populate with whatever sits in the database when not set by the token. This way we can only check one place (and we can probably speed up most of the permission-related checks by removing the database queries required to insert the permissions).

Reply:
I've changed the workflow. I dropped the changes in `permission_required` and extend our AuthBackend to use `effective_permissions`.